robot/rospackages/src/mcu_control/scripts/serial_util.py

```diff
@@ -0,0 +1,129 @@
+# setup serial communications by searching for arm teensy if USB, or simply connecting to UART
+# baud: baudrate, type: node type
+# baudrate & type:
+# PdsNode = 9600, 'PDS'
+# ArmNode = 115200, 'arm'
+# ScienceNode = 115200, 'science'
+# RoverNode = 115200, 'Astro'
+
+node_baudrate = {
+  "pds": 9600,
+  "arm": 115200,
+  "science": 115200,
+  "rover": 115200
+}
+
+node_type = {
+  "pds": 'PDS',
+  "arm": 'arm',
+  "science": 'science',
+  "rover": 'Astro'
+}
+
+def init_serial(baud, type):
+    baudrate = baud
+    # in a perfect world, you can choose the baudrate
+    rospy.loginfo('Using %d baud by default', baudrate)
+    # in a perfect world, usb vs uart will be set by ROS params
+    usb = False; uart = True
+    myargv = rospy.myargv(argv=sys.argv)
+    if len(myargv) == 1:
+        rospy.loginfo('Using UART by default')
+    if len(myargv) > 1:
+        if myargv[1] == 'uart':
+            usb=False; uart=True
+            if type = 'PDS':
+                rospy.loginfo('Using UART and 9600 baud by default')
+            else:
+                rospy.loginfo('Using UART and 115200 baud by default')
+        elif myargv[1] == 'usb':
+            usb=True; uart=False
+            if type = 'PDS':
+                rospy.loginfo('Using USB and 9600 baud by default')
```

Comment:
It's a bit awkward that you are logging that you are setting USB and saying that you set the baudrate but its actually set at a completely different location it seems